internal/server/web_server_e2e_test.go

```diff
@@ -0,0 +1,203 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/satsuma-data/node-gateway/internal/config"
+	"github.com/satsuma-data/node-gateway/internal/jsonrpc"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServeHTTP_ForwardsToSoleHealthyUpstream(t *testing.T) {
+	upstream := setUpHealthyUpstream(t, map[string]func(t *testing.T, w http.ResponseWriter){})
+	defer upstream.Close()
+
+	upstreamConfigs := []config.UpstreamConfig{
```

Comment:
Do we want to include one unhealthy node in this config?

Reply:
Added!